### PR TITLE
Launchpad: Add/new variable to click event

### PR DESCRIPTION
**Pull request by nelsonec87**
Related to https://github.com/Automattic/wp-calypso/issues/77616

## Proposed Changes

Added checklist complete status to the task click event. The idea is to be able to check the relevance of keeping the tasks available to users after the checklist is completed.

## Testing Instructions

* Create a new site from /start with the "Promote myself or business" intent.
* Launch your site
* You should see the new Launchpad task list on Customer Home.
* Run on your console: localStorage.debug='calypso:analytics'
* Click on a task
* Check data send on Devtools
* The task-clicked event should have a `checklist_completed` field

image

### CODE REVIEW

File: client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/sidebar.tsx [hunk 1]
```diff
@@ -212,7 +212,7 @@ const Sidebar = ( { sidebarDomain, siteSlug, submit, goNext, goToStep, flow }: S
 				) }
 				<Launchpad
 					siteSlug={ siteSlug }
-					taskFilter={ () => enhancedTasks }
+					taskFilter={ () => enhancedTasks || [] }
```
Comment (andres-blanco) [inline client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/sidebar.tsx:215]: Why is this change needed as part of this PR?

**Comment by sixhours** [comment]
Confirmed this is working as expected!

File: client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/sidebar.tsx (hunk 1 above)
Comment (nelsonec87) [reply client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/sidebar.tsx:215]: It is not directly related to the changes described in the PR. It's a small refactor in the spirit of _a leave-the-code-better-than-you-found-it_ .
I thought making the filter always return a `Task[]` would make the filter interface cleaner and easier to debug. Having to add `|| []` was a side effect.

**Review by andres-blanco — APPROVED** [review]
